www/js/app.js

```diff
@@ -786,9 +786,11 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
         $("#articleContent").contents().scrollTop(0);
 
         // Display the article inside the web page.
+        //Fast-replace img with data-img and hide image [kiwix-js #272]
+        htmlArticle = htmlArticle.replace(/(<img\s+[^>]*)src(\s*=)/ig,
+            "$1style=\"display: none;\" onload=\"this.style.display='inline'\" data-src$2");
```

Comment:
In general, jQuery certainly does the parsing better (in terms of compatibility) than we might do manually, even with a regexp. I'm thinking about old browser compatibility and/or the many ways to write HTML code.
For example, I'm not 100% sure what would happen if some style, onload or data-src attributes already existed in the img tag.
It seems jQuery can also work on HTML strings with https://api.jquery.com/jQuery.parseHTML/

But I have to say you did it very well, as you handle the case-sensitivity and multiline aspects.
Regarding speed, it's certainly much faster than a complete jQuery parsing of the String, that will be done again when the string will be inserted in the DOM with .html(htmlArticle).

Another idea would be to make jQuery parse the string with .parseHTML, change the img src attributes to data-src with jQuery, and then inject the elements in the body with .append instead of .html. So that the jQuery parsing is only done once (I did not test, though). The performance must be tested, but it might be similar, and we might use all the power of jQuery if we want to do more changes in the HTML before injecting it.